airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_25_0.java

```diff
@@ -0,0 +1,141 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.migrate.migrations;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.migrate.Migration;
+import io.airbyte.migrate.MigrationUtils;
+import io.airbyte.migrate.ResourceId;
+import io.airbyte.migrate.ResourceType;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+public class MigrationV0_25_0 extends BaseMigration implements Migration {
+
+  protected static final ResourceId DESTINATION_CONNECTION_RESOURCE_ID = ResourceId
+      .fromConstantCase(ResourceType.CONFIG, "DESTINATION_CONNECTION");
+  protected static final ResourceId STANDARD_SYNC_RESOURCE_ID = ResourceId
+      .fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC");
+  protected static final ResourceId STANDARD_SYNC_OPERATION_RESOURCE_ID = ResourceId
+      .fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC_OPERATION");
+
+  private static final String MIGRATION_VERSION = "0.25.0-alpha";
+  @VisibleForTesting
+  protected final Migration previousMigration;
+
+  public MigrationV0_25_0(Migration previousMigration) {
+    super(previousMigration);
+    this.previousMigration = previousMigration;
+  }
+
+  @Override
+  public String getVersion() {
+    return MIGRATION_VERSION;
+  }
+
+  private static final Path RESOURCE_PATH = Path.of("migrations/migrationV0_25_0/airbyte-config");
+
+  @Override
+  public Map<ResourceId, JsonNode> getOutputSchema() {
+    JsonNode schemaFromResourcePath = MigrationUtils.getSchemaFromResourcePath(RESOURCE_PATH, STANDARD_SYNC_OPERATION_RESOURCE_ID);
+
+    HashMap<ResourceId, JsonNode> resourceIdJsonNodeHashMap = new HashMap<>(previousMigration.getOutputSchema());
+    resourceIdJsonNodeHashMap.put(STANDARD_SYNC_OPERATION_RESOURCE_ID, schemaFromResourcePath);
+
+    return resourceIdJsonNodeHashMap;
+  }
+
+  @Override
+  public void migrate(Map<ResourceId, Stream<JsonNode>> inputData,
+                      Map<ResourceId, Consumer<JsonNode>> outputData) {
+
+    Set<String> destinationIds = new HashSet<>();
+
+    for (final Map.Entry<ResourceId, Stream<JsonNode>> inputEntry : inputData.entrySet()) {
+      if (!inputEntry.getKey().equals(DESTINATION_CONNECTION_RESOURCE_ID)) {
```

Comment:
it would be a bit easier for me to follow if you got rid of the for loop here and just did `inputData.getOrDefault(DESTINATION_CONNECTION_RESOURCE_ID, Stream.empty())`. This for loop is currently design to find 1 or 0 things.